File: assignment 10 - C# practice problems/26.cs
//Write a c# function to convert a binary number to a decimal number.

using System;
namespace MyNameSpace
{
    class Program26
    {
        public static void BinaryToDecimalDriverFunction()
        {

            Console.WriteLine("Enter a binary Number: ");
            var inputNumber = int.Parse(Console.ReadLine());

            Console.WriteLine("The decimal equivalant is " + binaryToDecimal(inputNumber));

        }

        static int binaryToDecimal(int numberParam)
        {

            var decimalResult = 0;
            var power = 0;
            while (numberParam > 0)
            {
                decimalResult += (int)((numberParam % 10) * Math.Pow(2, power));
                numberParam = (int)(numberParam / 10);
                ++power;
            }

            return decimalResult;
        }

    }
}

File: assignment 3 - js practice problems/web-dev-starter/26.js
//Write a JavaScript function to convert a binary number to a decimal number.
function binaryToDecimal(numberParam) {

    let decimal = 0;
    let power = 0;
    while (numberParam > 0) {
        decimal += ((numberParam % 10) * Math.pow(2, power));
        numberParam = parseInt(numberParam / 10);
        ++power;
    }

    return decimal;
}

function binaryToDecimalDriverFunction() {
    let numberInput = prompt("Enter a binary Number");

    alert("The decimal equivalant is " + binaryToDecimal(numberInput));
}

binaryToDecimalDriverFunction();